docs/core-concepts/app-types.md

```diff
@@ -0,0 +1,125 @@
+
+1. Portable application
+    * As a sub-category of this we have the portable application with native dependencies
+2. Self-contained application
+
+## Portable application
+Portable applications are the default type in .NET Core. They require .NET Core to be installed on the targeted machine 
+in order for them to run. To you as a developer, this means that your application is portable between installations of 
+.NET Core. 
+
+This type of application will only carry its own code and dependencies that are outside of .NET Core libraries. 
+As an example, let's say you are making a console application that has the ability to invoke a certain REST API 
+and deserialize the returned JSON into a type and then display it. You have everything you need for this small 
+application expect for a good JSON parser; for this, you add a dependency to your `project.json` to include 
+[Json.NET](https://www.nuget.org/packages/Newtonsoft.Json/). Once you publish your application using `dotnet publish`, 
+you will see that only your application's code and JSON.net have been published in the output. 
+The .NET Core libraries remains outside of your application's dependency closure. 
+
+In order to create a portable application, all you need to do is to target the .NET Core libraries in your `project.json`
+and have your frameworks aligned as the below sample shows. 
+
+```json
+"dependencies": {
+    "Microsoft.NETCore.App": {  
+        "version": "1.0.0-*",
+        "type": "platform"
+    }
+},
+"frameworks": {
+    "netcoreapp1.0": {}
+}
+```
+
+The `Microsoft.NETCore.App` is a "meta-package" that states that you are targeting the .NET Core libraries. The `type: platform` 
+property on that dependency means that at publish time, the tooling will skip publishing the assemblies for that dependency 
+to the published output. You don't need these since they will be installed with .NET Core on the targeted machine. 
+
+### Portable application with native dependencies
+A sub-group of the above, this type is a portable application that has native dependencies specified 
+somewhere in its dependency chain. This application is as portable as all of its native dependencies 
+are portable. You will be able to run the application on any platform that your native dependencies can 
+run on. Prime example of this is Kestrel, the ASP.NET Core cross-platform web server. It is built on top of 
+[libuv](https://github.com/libuv/libuv) which is its native dependency. 
+
+When you publish a portable application that has a native dependency, the published output will contain 
+all the same things as the portable application described in the previous section. For native dependencies
+the published output will contain a folder for each [RID](#what-are-rids) that the native dependency supports 
+(and that exists in its NuGet package). 
+
+As said, the portable application with a native dependency is the same as a portable application, just with a 
```

Comment:
@mairaw if is slightly more involved than that. It is not just about adding native dependencies but also what those mean for your portability.